presto-docs/src/main/sphinx/connector/kafka.rst

```diff
@@ -259,6 +264,430 @@ Field           Required  Type      Description
+
+The following field attributes are supported:
+
+* ``dataFormat`` - Selects the width of the column data type
+* ``type`` - Presto data type
+* ``mapping`` - ``<start>[:<end>]``; start and end position of bytes to convert
+
+The ``dataFormat`` attribute selects the number of bytes converted.
+If absent, ``BYTE`` is assumed. All values are signed.
+
+Supported values are:
+
+* ``BYTE`` - one byte
+* ``SHORT`` - two bytes (big-endian)
+* ``INT`` - four bytes (big-endian)
+* ``LONG`` - eight bytes (big-endian)
+* ``FLOAT`` - four bytes (IEEE 754 format)
+* ``DOUBLE`` - eight bytes (IEEE 754 format)
+
+The ``type`` attribute defines the Presto data type.
+
+Depending on Presto type assigned to column different values of ``dataFormat``
+can be used:
+
+===================================== =======================================
+Presto data type                      ``dataFormat`` values
+===================================== =======================================
+``BIGINT``                            ``BYTE``, ``SHORT``, ``INT``, ``LONG``
+``INTEGER``                           ``BYTE``, ``SHORT``, ``INT``
+``SMALLINT``                          ``BYTE``, ``SHORT``
+``TINYINT``                           ``BYTE``
+``DOUBLE``                            ``DOUBLE``, ``FLOAT``
+``BOOLEAN``                           ``BYTE``, ``SHORT``, ``INT``, ``LONG``
+``VARCHAR`` / ``VARCHAR(x)``          ``BYTE``
+===================================== =======================================
+
+The ``mapping`` attribute specifies the range of bytes in a key or
+message used for encoding. It can be one or two numbers separated by a colon
+(``<start>[:<end>]``).
+
+.. note::
+
+    For ``VARCHAR`` types a start and end position must be defined. Otherwise
+    there is no way to know how many bytes the message contains. The raw format
+    mapping information is static and cannot be dynamically changed to fit the
+    variable width of some Presto data types.
+
+If only a start position is given:
+
+* For fixed width types the appropriate number of bytes are used for the
+  specified ``dateFormat`` (see above).
+
+If start and end position are given, then:
+
+* For fixed width types the size must be equal to number of bytes used by
+  specified ``dataFormat``.
+* For ``VARCHAR`` all bytes between start (inclusive) and end (exclusive) are
+  used.
+
+.. note::
+
+    At the very least a start value must be present in the mapping for encoding
+    to work.
+
+Encoding scheme of numeric data types (``BIGINT``, ``INTEGER``, ``SMALLINT``,
+``TINYINT``, ``DOUBLE``) is straightforward. Table column values are encoded to
+bytes according to either:
+
+* big-endian encoding (for integer types)
+* IEEE 754 format for (for ``DOUBLE``).
+
+Example raw field definition in a :ref:`table definition file<kafka-table-definition-files>`
+for a Kafka message:
+
+.. code-block:: none
+
+    "message": {
+            "dataFormat": "raw",
```

Comment:
The indentation seems off here. Let's use two-space indent to be consistent with other documentation.